app/src/main/java/nu/yona/app/ui/LaunchActivity.java

```diff
@@ -213,14 +213,40 @@ private void setListeners()
 
 	private void validateYonaPasswordEncryption()
 	{
+		int currentEncryptionMode = getSharedUserPreferences().getInt(YONA_ENCRYPTION_METHOD, EncryptionMethod.INITIAL_METHOD.ordinal());
```

Comment:
Declare this as the ``EncryptionMethod`` enum, using the conversion method as explained [here](https://stackoverflow.com/a/609866/4353482).

Reply:
Done.